src/main/java/picard/illumina/IlluminaBasecallsToSam.java

```diff
@@ -455,12 +465,20 @@ private void populateWritersFromLibraryParams() {
         final Map<String, String> params = new LinkedHashMap<>();
 
         String platformUnit = RUN_BARCODE + "." + LANE;
-        if (barcodes != null) platformUnit += ("." + IlluminaUtil.barcodeSeqsToString(barcodes));
+        if (barcodes != null) {
+            final String barcodeString = IlluminaUtil.barcodeSeqsToString(barcodes);
+            platformUnit += ("." + barcodeString);
+            // TODO: replace with htsjdk constant after rev
+            if (INCLUDE_BC_IN_RG_TAG) params.put("BC", barcodeString);
```

Comment:
please format `if` body across lines with `{` `}` here and below on line 480 below.